packages/url_launcher/url_launcher_platform_interface/lib/url_launcher_platform_interface.dart

```diff
@@ -4,27 +4,50 @@
 
 import 'dart:async';
 
-import 'package:meta/meta.dart' show required;
+import 'package:meta/meta.dart' show required, visibleForTesting;
 
 import 'method_channel_url_launcher.dart';
 
 /// The interface that implementations of url_launcher must implement.
 ///
-/// Platform implementations that live in a separate package should extend this
-/// class rather than implement it as `url_launcher` does not consider newly
-/// added methods to be breaking changes. Extending this class (using `extends`)
-/// ensures that the subclass will get the default implementation, while
-/// platform implementations that `implements` this interface will be broken by
-/// newly added [UrlLauncherPlatform] methods.
+/// Platform implementations should extend this class rather than implement it as `url_launcher`
+/// does not consider newly added methods to be breaking changes. Extending this class
+/// (using `extends`) ensures that the subclass will get the default implementation, while
+/// platform implementations that `implements` this interface will be broken by newly added
+/// [UrlLauncherPlatform] methods.
 abstract class UrlLauncherPlatform {
+  /// Only mock implementations should set this to true.
+  ///
+  /// Mockito mocks are implementing this class with `implements` which is forbidden for anything
+  /// other than mocks (see class docs). This property provides a backdoor for mockito mocks to
+  /// skip the verification that the class isn't implemented with `implements`.
+  @visibleForTesting
+  bool get isMock => false;
+
   /// The default instance of [UrlLauncherPlatform] to use.
   ///
   /// Platform-specific plugins should override this with their own
   /// platform-specific class that extends [UrlLauncherPlatform] when they
   /// register themselves.
   ///
   /// Defaults to [MethodChannelUrlLauncher].
-  static UrlLauncherPlatform instance = MethodChannelUrlLauncher();
+  static UrlLauncherPlatform _instance = MethodChannelUrlLauncher();
+
+  static UrlLauncherPlatform get instance => _instance;
+
+  // TODO(amirh): Extract common platform interface logic.
+  // https://github.com/flutter/flutter/issues/43368
+  static set instance(UrlLauncherPlatform instance) {
```

Comment:
nit: style guide gives the order for getters/setters as "getter/field/setter" with no whitespace between them